src/apify_shared/utils.py

```diff
@@ -0,0 +1,106 @@
+import io
+import json
+import re
+from datetime import datetime, timezone
+from enum import Enum
+from typing import Any, Dict, List, Optional, TypeVar, cast
+
+PARSE_DATE_FIELDS_MAX_DEPTH = 3
+PARSE_DATE_FIELDS_KEY_SUFFIX = 'At'
+
+ListOrDict = TypeVar('ListOrDict', List, Dict)
+T = TypeVar('T')
+
+
+def filter_out_none_values_recursively(dictionary: Dict) -> Dict:
+    """Return copy of the dictionary, recursively omitting all keys for which values are None."""
+    return cast(dict, filter_out_none_values_recursively_internal(dictionary))
+
+
+def filter_out_none_values_recursively_internal(
```

Comment:
I think this method could be prefixed with `_`, to make it explicit that it's private and internal.